packages/framework-core/src/services/graphql/graphql-type-informer.ts

```diff
@@ -47,11 +47,12 @@ export class GraphQLTypeInformer {
 
       if (primitiveType === Array) {
         const param = prop.typeInfo.parameters[0]
-        const graphQLPropType = this.getGraphQLTypeFor(param.type)
+        let graphQLPropType = this.getGraphQLTypeFor(param.type)
 
         if (!this.isGraphQLScalarType(graphQLPropType) && param.type) {
           const properties = getPropertiesMetadata(param.type)
           this.generateGraphQLTypeFromMetadata({ class: param.type, properties })
+          graphQLPropType = this.getGraphQLTypeFor(param.type)
```

Comment:
Isn't this line (55) doing the same as line 50? Is this call be redundant maybe?

Reply:
It's already a while ago but this "duplication" was on purpose. If I'm correct the main problem is `generateGraphQLTypeFromMetadata` which adds information into a map, information which is needed by `getGraphQLTypeFor` to generate the correct type. Without the second `getGraphQLTypeFor`, some types result in JSONObject. 

It's not the most elegant solution, it's rather a fix as larger restructuring will probably take place related to https://github.com/boostercloud/metadata-booster/pull/2.

Reply:
Oh, got it, then this is definitely calling for restructuring, having methods mutating stuff gives me goosebumps...  Thanks for the explanation!